visualization/app/codeCharta/ui/dialog/dialog.service.ts

```diff
@@ -38,26 +38,45 @@ export class DialogService {
 		await this.$mdDialog.show(this.$mdDialog.alert().clickOutsideToClose(true).title(title).htmlContent(message).ok(button))
 	}
 
-	async showErrorDialogAndOpenFileChooser(message = "An error occurred.", title = "Error", button = "Ok") {
-		const prompt = this.$mdDialog.alert().clickOutsideToClose(true).title(title).htmlContent(message).ok(button)
-		await this.$mdDialog.show(prompt)
-		document.getElementById("input-file-id").click()
+	async showValidationDialog(fileValidationResults: CCFileValidationResult[]) {
+		const htmlMessages = []
+
+		if (fileValidationResults.flatMap(validation => validation.errors).length > 0) {
+			htmlMessages.push("<h2>Errors</h2>")
+			for (const fileValidationResult of fileValidationResults) {
+				if (fileValidationResult.errors.length > 0) {
+					const fileErrorMessage = this.buildFileErrorMessage(fileValidationResult)
+					htmlMessages.push(fileErrorMessage)
+				}
+			}
+		}
```

Comment:
I would not iterate twice over the validation results. Here a suggestion to save one iteration:
```suggestion
		const errors = fileValidationResults.filter((validationResult) => { return validationResult.errors.length > 0 })
		if (errors .length > 0) {
			htmlMessages.push("<h2>Errors</h2>")
			for (const error of errors ) {
					const fileErrorMessage = this.buildFileErrorMessage(error )
					htmlMessages.push(fileErrorMessage)
			}
		}
```

Reply:
Doesn't filter iterate over the array too? Or is it less expansive than flatmap?

Reply:
Yes, it does. But your second loop in line 46 will iterate all validation results again (instead of just the errors). Furthermore, it must be checked if there are any errors present a second time  .

Reply:
Resolved in the next commit